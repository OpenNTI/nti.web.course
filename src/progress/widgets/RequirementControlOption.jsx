import React from 'react';
import PropTypes from 'prop-types';

export default class RequirementControlOption extends React.Component {
	static propTypes = {
		option: PropTypes.object.isRequired,
		onChange: PropTypes.func,
		isSelected: PropTypes.bool
	}

	onClick = (e) => {
		e.preventDefault();
		e.stopPropagation();

		const {onChange} = this.props;

		if(onChange) {
			onChange(this.props.option.value);
		}
	}

	render () {
		return (
			<div className="require-control-option" onClick={this.onClick}>
				{this.props.isSelected && <i className="icon-check"/>}
				{this.props.option.label}
			</div>
		);
	}
}
