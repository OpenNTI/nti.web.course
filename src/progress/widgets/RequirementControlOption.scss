@import "~nti-style-common";

.require-control-option {
	padding-left: 36px;
	position: relative;
	height: 42px;
	line-height: 42px;
	cursor: pointer;

	i {
		position: absolute;
		left: 12px;
		top: 12px;
		color: $primary-blue;
		font-size: 18px;
	}
}
