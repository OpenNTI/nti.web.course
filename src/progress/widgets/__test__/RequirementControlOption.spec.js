import React from 'react';
import renderer from 'react-test-renderer';

import RequirementControlOption from '../RequirementControlOption';

/* eslint-env jest */
describe('Progress widgets requirement control option test', () => {
	test('Not selected', async () => {
		const cmp = renderer.create(
			<RequirementControlOption option={{ label: 'Option 1' }} />
		);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Selected', async () => {
		const cmp = renderer.create(
			<RequirementControlOption
				option={{ label: 'Option 1' }}
				isSelected
			/>
		);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
