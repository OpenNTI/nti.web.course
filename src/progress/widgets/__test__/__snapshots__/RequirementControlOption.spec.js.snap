// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress widgets requirement control option test Not selected 1`] = `
<div
  className="require-control-option"
  onClick={[Function]}
>
  Option 1
</div>
`;

exports[`Progress widgets requirement control option test Selected 1`] = `
<div
  className="require-control-option"
  onClick={[Function]}
>
  <i
    className="icon-check"
  />
  Option 1
</div>
`;
